updating useAsyncStorage syntax

idkjs opened the pull request:
Lets user do:

#### `ReactNativeAsyncStorage.asyncStorageState`

```reason
type asyncStorageState = {
  getItem: unit => Js.Promise.t(Js.Null.t(string)),
  setItem: string => Js.Promise.t(unit),
  mergeItem: string => Js.Promise.t(unit),
  removeItem: unit => Js.Promise.t(unit),
};

let key = "@@Key"
 let ReactNativeAsyncStorage.{getItem, setItem} =
    ReactNativeAsyncStorage.useAsyncStorage("@@Key");
```

Instead of 

```reason
 let useAsyncStorage=
    ReactNativeAsyncStorage.useAsyncStorage("@@Key");

let getItem =useAsyncStorage##getItem;
```

See updated [README.md](https://github.com/reason-react-native/async-storage/blob/1678c86/README.md#L49)

idkjs committed:
updating useAsyncStorage syntax

idkjs committed:
Update README.md

tatchi commented on README.md:
```diff
@@ -131,6 +134,104 @@ unit => unit
+  let ReactNativeAsyncStorage.{getItem, setItem} =
+    ReactNativeAsyncStorage.useAsyncStorage("@@KEY");
+
+  let getItem = () => useAsyncStorage##getItem();
```
Shouldn't this line be removed?

idkjs replied:
Yes. Done. Thank you.

idkjs committed:
Update Readme per comment

https://github.com/reason-react-native/async-storage/pull/4#discussion_r529743105

idkjs committed:
Add Rescript example

MoOx reviewed (changes requested):
Thanks for this PR! Could the example be more straight forward without styles for example? To avoid unecessary noise so the developer can focus on the actual interesting code, thanks!

idkjs committed:
remove style from example per https://github.com/reason-react-native/async-storage/pull/4#pullrequestreview-564167837

idkjs commented:
@MoOx https://github.com/reason-react-native/async-storage/pull/4#pullrequestreview-564167837 done

MoOx committed:
remove complex examples

MoOx commented:
I removed example as they were way to complex to quickly illustrate how to use the hook. If you want to another PR, but without UI & faker & unecessary stuff for the developer to understand the core of the usage.

MoOx commented:
I rolled back the change and added record syntax as `useAsyncStorage_` with a deprecation notice for old syntax.
